src/marlin_stubs/host/M115.cpp

```diff
@@ -39,18 +40,26 @@ static void cap_line(PGM_P const name, bool ena = false) {
  * M115: Capabilities string
  */
 void GcodeSuite::M115() {
-
+    uint8_t extruders = 0;
+    char extruder[4] = { 0 };
     SERIAL_ECHOPGM("FIRMWARE_NAME:Prusa-Firmware-Buddy ");
     SERIAL_ECHOPGM(project_version_full);
     SERIAL_ECHOPGM(" (Github) SOURCE_CODE_URL:https://github.com/prusa3d/Prusa-Firmware-Buddy");
-#if PRINTER_IS_PRUSA_MK4
-    SERIAL_ECHOLNPGM(" PROTOCOL_VERSION:" PROTOCOL_VERSION " MACHINE_TYPE:"
-                     "MK4"
-                     " EXTRUDER_COUNT:" STRINGIFY(1) " UUID:" MACHINE_UUID); // extruder hardcoded due to mmu, should be dynamic in the future
+    SERIAL_ECHOPGM(" PROTOCOL_VERSION:" PROTOCOL_VERSION " MACHINE_TYPE:" PRINTER_MODEL
+                   " EXTRUDER_COUNT:");
+#ifdef HAS_MMU2
+    // Devices with MMU2/MMU3 only have one physical extruder, whereas EXTRUDERS refers to total amount of virtual + physical extruders
+    // Example: MK4 (EXTRUDERS: 1) / MK4 with MMU3 (EXTRUDERS: 5)
+    // As only physical extruders are reported here, we hardcode the value to 1
+    extruders = 1;
+#elif HAS_TOOLCHANGER()
+    extruders = PrusaToolChangerUtils::get_num_enabled_tools();
 #else
-    SERIAL_ECHOLNPGM(" PROTOCOL_VERSION:" PROTOCOL_VERSION " MACHINE_TYPE:" MACHINE_NAME " EXTRUDER_COUNT:" STRINGIFY(EXTRUDERS) " UUID:" MACHINE_UUID);
+    extruders = EXTRUDERS;
 #endif
-
+    snprintf(extruder, sizeof(extruder), "%d", extruders);
+    SERIAL_ECHOPGM(extruder);
```

Comment:
`PGM` stands for "program memory". `extruder` is not in the program memory, so using this function is not valid.

Reply:
You also don't need to use the snprintf at all, there is the `SERIAL_ECHO_F` function. To make the code readable, I'd do `SERIAL_ECHO_F(" EXTRUDER_COUNT:%d", extruder_count);`